File: Task-2/task2_ws/src/turtle/src/pub_twist.cpp
#include <ros/ros.h>
#include <geometry_msgs/Twist.h>

using namespace ros;

int main(int argc, char **argv)
{
    // node
    init(argc, argv, "pub_twist", init_options::AnonymousName);
    NodeHandle nh;

    // publisher
    Publisher pub = nh.advertise<geometry_msgs::Twist>("turtle1/cmd_vel", 1);
    Rate rate(1);

    // message
    geometry_msgs::Twist twist;
    twist.angular.z = 1;
    twist.linear.x = 1;

    // publishing
    while (ok())
    {
        pub.publish(twist);
        ROS_INFO("\ntwist.angular.z = %f\ntwist.linear.x = %f",twist.angular.z,twist.linear.x);
        spinOnce();
        rate.sleep();
    }

    return 0;
}

File: Task-2/task2_ws/src/turtle/scripts/pub_twist.py
#!/usr/bin/python3

import rospy
from geometry_msgs.msg import Twist


def pub_twist():

    # node
    rospy.init_node("pub_twist", anonymous=True)

    # publisher
    pub = rospy.Publisher("turtle1/cmd_vel", Twist, queue_size=1)
    rate = rospy.Rate(1)

    # message
    twist = Twist()
    twist.angular.z = 1
    twist.linear.x = 1

    # publishing
    while not rospy.is_shutdown():
        pub.publish(twist)
        rospy.loginfo(twist)
        rate.sleep()


if __name__ == "__main__":
    try:
        pub_twist()
    except rospy.ROSInterruptException:
        pass
